### PR TITLE
test: Add test to return correct field values

### PR DESCRIPTION
**Pull request by mtrezza**
This test is to ensure that server-side changes like https://github.com/parse-community/parse-server/pull/9115 don't cause issues like https://github.com/parse-community/Parse-SDK-JS/issues/2198. Same test added to JS SDK in https://github.com/parse-community/Parse-SDK-JS/pull/2201.

### CODE REVIEW

**Comment by dplewis** [comment]
This will need the next SDK release to work.

**Comment by mtrezza** [comment]
Yes, I'll release a stable version of the JS SDK later today and then merge that in Parse Server -- done as v5.3.0

File: spec/ParseObject.spec.js [hunk 1]
```diff
@@ -2055,4 +2055,42 @@ describe('Parse.Object testing', () => {
...
+      { field: 'object', value: { '0': 0 } },
+      { field: 'object', value: { '1': 1 } },
+      { field: 'object', value: { '0': { '0': 'a', '1': 'b' } } },
+      { field: 'date', value: new Date() },
+      {
+        field: 'file',
+        value: Parse.File.fromJSON({
+          __type: 'File',
+          name: 'name',
+          url: 'http://localhost:1337/parse/files/integration/name',
```
Comment (mtrezza) [inline spec/ParseObject.spec.js]:
```suggestion
          url: 'http://localhost:8378/1/files/test/name',
```

Fix error:
  - Expected $._url = 'http://localhost:8378/1/files/test/name' to equal 'http://localhost:1337/parse/files/integration/name'.

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.1.0-alpha.15](https://github.com/parse-community/parse-server/releases/tag/7.1.0-alpha.15)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.2.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/7.2.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.2.0](https://github.com/parse-community/parse-server/releases/tag/7.2.0)

**Pull request merged**